enhancements/installer/compact-plus-worker.md

```diff
@@ -0,0 +1,194 @@
+  cluster as a single installation step so that my cluster will be able to
+  handle workloads that require replication.
+
+
+### API Extensions
+
+- Sets the `infrastructureTopology` field to the value of`ControlPlaneTopology` in the
+  `config.openshift.io/v1/infrastructure` CR called `cluster`, which is created
+  during the install process, when the number of workers is 1.
+
+- Sets the `mastersSchedulable` field to true in the
+  `config.openshift.io/v1/scheduler` CR called `cluster`, which is created
+  during the install process, when the number of workers is 1.
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+### Risks and Mitigations
+
+This should not introduce newer risks other than the existing ones in a compact
+cluster deployment.
+
+## Design Details
+
+### Open Questions [optional]
+
+Nothing at the moment
+
+### Test Plan
+
+
+- Add unit tests to ensure the Infrastructure and Scheduler settings are properly set for the different combinations
+
+- Add a periodic job (nightly?) job to run OpenShift tests on a 4-node cluster
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+TBD
+
+#### Dev Preview -> Tech Preview
+
+TBD
+
+#### Tech Preview -> GA
+
+TBD
+
+#### Removing a deprecated feature
+
+TBD
+
+### Upgrade / Downgrade Strategy
+
+This enhancement does not have an impact on existing clusters as it's intended
+to affect only day0 operations.
+
+### Version Skew Strategy
+
+Does not apply
+
+### Operational Aspects of API Extensions
+
+
+- Allows for a 4-nodes, highly available, cluster to be deployed as a day0
+  task. It reduces the time required for such deployments to be created.
+
+- Does not impact general deployment functionalities in a negative way.
+
+- Allowing workloads on the master nodes may affect the master's health if the
+  traffic and workloads are too high.
+
+#### Failure Modes
+
+Same as a compact cluster
+
+#### Support Procedures
+
+- The same support procedures used for a compact cluster should be used for a
+  4-nodes cluster. I would appreciate help filling in this section with more
+  details.
+
+
+## Implementation History
+
+N/A
+
+## Drawbacks
+
+A 4-nodes cluster (a.k.a 3+1) is generally not a recommended architecture. By
+having schedulable masters users will be routing ingress traffic through the
+masters as well as other workloads. Depending on the amount of traffic, and the
+resources available to the cluster, this may result in degraded functionality
+in the masters, which finally affects the general health of the cluster.
+
+With this enhancement, we will be making it easier for users to create such
+4-nodes cluster deployments, which could make users believe it is supported or
```

Comment:
If it is not supported, then why are we adding ways to explicitly make it easier to do?

Reply:
Goes back to the point I made in the `non-goals` section. To be honest, it's unclear if this architecture is really supported or not. 

The installer won't let you do  it in one go due to the issues mentioned in this enhancement but users can still get a 4-nodes cluster in two steps. It's unclear where the "supportability" of a topology is defined

Reply:
> Supportability means that if a user performs the actions and ends up having a problem, then we will help them fix the problem without first requiring that they undo the actions. The fact that it takes the user two steps to get there has no bearing on whether it is something that we do or do not support.

I will remove references of the term "support" in this enhancement. 

However, I think it's more complicated than how you are phrasing it here. There's always a gap between what developers think is supported and what the user thinks is supported. Many of the actions we take will communicate to the user whether we think something is ok or not. Finally, our actions in cases like this will help us provide better support (to @hardys point on what the test matrix would look like).

Reply:
Do you think that if we implemented this EP that we would no longer support the 3+1 configuration where the control planes nodes are not schedulable for workloads? If so, should we actively reject that configuration in-cluster?